test/unit/logic/provider_upgrade_test.rb

```diff
@@ -6,19 +6,21 @@ def setup
     service = master_account.first_service!
     @provider = FactoryBot.create(:provider_account)
     @power1M = FactoryBot.create(:published_plan, :system_name => 'power1M', :issuer => service)
-    @power1M.plan_rule.stubs(:switches).returns(%i[finance multiple_applications branding require_cc_on_signup
+    rule = @power1M.send :plan_rule
+    rule.stubs(:switches).returns(%i[finance multiple_applications branding require_cc_on_signup
       account_plans multiple_users groups]
     )
-    @power1M.plan_rule.stubs(:limits).returns(PlanRule::Limit.new(max_services: 1, max_users: 1))
-    @power1M.plan_rule.stubs(:rank).returns(10)
+    rule.stubs(:limits).returns(PlanRule::Limit.new(max_services: 1, max_users: 1))
+    rule.stubs(:rank).returns(10)
     @pro = FactoryBot.create(:published_plan, :system_name => 'pro3M', :issuer => service)
-    @pro.plan_rule.stubs(:switches).returns(%i[finance multiple_applications branding require_cc_on_signup account_plans
+    rule = @pro.send :plan_rule
```

Comment:
why reusing the name for this variable? Why not use a different one? it would be less confusing.

Reply:
Just saving in a local variable to avoid duplicate calls. Suggestions for naming?

Reply:
`pro_rule` ?

Reply:
fixed